feat(create-form-field-description-css): add form field description css package (#1241)
nl-design-system/candidate: packages/components-react/form-field-description-react/README.md

```shell
pnpm add @nl-design-system-candidate/form-field-description-react
```

yarn:

```shell
yarn add @nl-design-system-candidate/form-field-description-react
```

## Usage

With built-in CSS:

```jsx
import { FormFieldDescription } from '@nl-design-system-candidate/form-field-description-react/css';

<FormFieldDescription id="name-description">
  Enter your full name, as shown on your identity document.
</FormFieldDescription>;
```

Only the React component. You still need to import the CSS separately:

```jsx
import { FormFieldDescription } from '@nl-design-system-candidate/form-field-description-react';

<FormFieldDescription id="name-description">
  Enter your full name, as shown on your identity document.
</FormFieldDescription>;
```

### Linking to an input field via `aria-describedby`

Always give the `FormFieldDescription` an `id` and link it via `aria-describedby` to the associated input field. This allows screen readers to read out the description when the input field receives focus.

```jsx
<div>
  <label htmlFor="naam">Name</label>
  <FormFieldDescription id="name-description">
    Enter your full name, as shown on your identity document.
  </FormFieldDescription>

>   <input id="naam" aria-describedby="name-description" type="text" />

Reviewer: ```suggestion
  <label htmlFor="name">Name</label>
  <FormFieldDescription id="name-description">
    Enter your full name, as shown on your identity document.
  </FormFieldDescription>
  <input id="name" aria-describedby="name-description" type="text" />
```

Author: Done

# Form Field Description

> Toont ondersteunende tekst en biedt extra context over de in te vullen informatie.

Reviewer: Ik denk dat we hier nog een officiele component beschrijving voor willen maken, en later deze tekst moeten vervangen door de betere tekst.

@AliKdhim87 wil jij hiervoor een issue aanmaken die gekoppeld is aan de tracking issue van Form Field Description?

@AliKdhim87 wil jij hiervoor met Charlotte overleggen of we hier een stap voor missen in de Definition of Done voor Candidate? Ik heb het gevoel dat we er vanuit gaan dat de officele beschrijving al goedgekeurd was door de Design System Lead, en dat lijkt niet te kloppen.

Author: Ik heb het [issue](https://github.com/nl-design-system/candidate/issues/1316) aangemaakt

Reviewer 2: Vraag van Robbert beantwoord: 
- De description word ingevuld in de packages stap van Keywords en Aliassen https://github.com/nl-design-system/candidate/blob/main/.github/ISSUE_TEMPLATE/204-%F0%9F%93%9D-keywords-en-aliassen.md#maak-de-packages-beschikbaar-voor-de-nieuwe-candidate-component
- Er is geen Candidate proces stap die de omschrijving van de component bedenkt; we nemen de omschrijving over van de website / huidige omschrijving die waarschijnlijk uit Help Wanted kwam.
- Ik zal met Yolijn bespreken of dit een stap in Candidate moet zijn; bijvoorbeeld verifieer component omschrijving.

Reviewer: @AliKdhim87 Bedankt voor het aanmaken van de issue! Je had 'm nog niet gekoppeld aan de tracking issue, maar dat heb ik nu wel gedaan: https://github.com/nl-design-system/candidate/issues/939

## Installatie

npm:

```shell
npm install @nl-design-system-candidate/form-field-description-react
```

pnpm:

```shell
pnpm add @nl-design-system-candidate/form-field-description-react
```

yarn:

```shell
yarn add @nl-design-system-candidate/form-field-description-react
```

## Gebruik

Met ingebouwde CSS:

```jsx
import { FormFieldDescription } from '@nl-design-system-candidate/form-field-description-react/css';

> <FormFieldDescription id="naam-description">

Reviewer: Zou je de `id` en `htmlFor` enzo helemaal Engels kunnen houden? Dus bijv: `name-description`?

(In alle voorbeelden)

  Voer je volledige naam in, zoals vermeld op je identiteitsbewijs.
</FormFieldDescription>;
```

> Alleen de React component, CSS apart importeren:

Reviewer: ```suggestion
Alleen de React component. De CSS moet je nog apart importeren:
```

```jsx
import { FormFieldDescription } from '@nl-design-system-candidate/form-field-description-react';

<FormFieldDescription id="naam-description">
  Voer je volledige naam in, zoals vermeld op je identiteitsbewijs.
</FormFieldDescription>;
```

### Koppelen aan een invoerveld via `aria-describedby`

Geef de `FormFieldDescription` altijd een `id` en koppel dit via `aria-describedby` aan het bijbehorende invoerveld. Zo kunnen screenreaders de beschrijving voorlezen wanneer het invoerveld focus krijgt.

```jsx
<label htmlFor="naam">Naam</label>
<FormFieldDescription id="naam-description">
  Voer je volledige naam in, zoals vermeld op je identiteitsbewijs.
</FormFieldDescription>

> <input id="naam" aria-describedby="naam-description" type="text" />

Reviewer: ```suggestion
<div>
  <label htmlFor="naam">Naam</label>
  <FormFieldDescription id="naam-description">
    Voer je volledige naam in, zoals vermeld op je identiteitsbewijs.
  </FormFieldDescription>
  <input id="naam" aria-describedby="naam-description" type="text" />
</div>
```

Reviewer: <img width="620" height="249" alt="Image" src="https://github.com/user-attachments/assets/53312f3b-6e24-4401-a944-346e3194e471" />

De syntax highlighting ziet er een beetje crappy uit doordat het niet gebalanceerde JSX is, zullen we er een divje omheen zetten?

```

### Disabled state

Gebruik de `disabled` prop wanneer het bijbehorende formulierveld uitgeschakeld is.

```jsx
<FormFieldDescription id="naam-description" disabled>

>   Dit veld is momenteel niet beschikbaar.

Reviewer: Laten we hier ook een realistisch voorbeeld blijven gebruiken.

```suggestion
  Voer je volledige naam in, zoals vermeld op je identiteitsbewijs.
```

</FormFieldDescription>
```

### Inhoud van de Form Field Description

De `FormFieldDescription` accepteert platte tekst en phrasing content als `children`, bijvoorbeeld een `<p>` element of inline elementen zoals `<strong>` en `<em>`. Gebruik geen interactive content (zoals links of knoppen) als `children`.

```jsx
<FormFieldDescription id="naam-description">
  <p>Voer je volledige naam in.</p>
</FormFieldDescription>
```

## Props

| Prop       | Beschrijving                                                                                | Type        | Standaard |
| ---------- | ------------------------------------------------------------------------------------------- | ----------- | --------- |

> | `id`       | Koppelt de beschrijving via `aria-describedby` aan een invoerveld. Gebruik altijd een `id`. | `string`    | —         |

Reviewer: ```suggestion
| `id`       | Koppel de beschrijving via `aria-describedby` aan een invoerveld. Gebruik altijd een `id`. | `string`    | —         |
```

| `children` | Inhoud van de form field description                                                        | `ReactNode` | —         |
| `disabled` | Geeft aan dat het bijbehorende formulierveld uitgeschakeld is                               | `boolean`   | `false`   |

Naast bovenstaande props accepteert `FormFieldDescription` alle standaard HTML attributen van een `<div>` element.

## Design Tokens

> Alle NL Design System componenten zijn white label. Gebruik de Form Field Description design tokens om de stijl aan te passen aan jouw huisstijl. Zie voor meer informatie over design tokens [https://nldesignsystem.nl/handboek/huisstijl/design-tokens/](https://nldesignsystem.nl/handboek/huisstijl/design-tokens/).

Reviewer: ```suggestion
Alle NL Design System componenten zijn white-label. Gebruik de Form Field Description design tokens om de stijl aan te passen aan jouw huisstijl. Zie voor meer informatie over design tokens [https://nldesignsystem.nl/handboek/huisstijl/design-tokens/](https://nldesignsystem.nl/handboek/huisstijl/design-tokens/).
```

De tokens voor Form Field Description zijn te vinden in het tokens pakket [`@nl-design-system-candidate/form-field-description-tokens`](https://www.npmjs.com/package/@nl-design-system-candidate/form-field-description-tokens).

## Andere implementaties

> Wil je Form Field Description gebruiken zonder React? De React component is gebaseerd op

Reviewer: ```suggestion
Wil je Form Field Description gebruiken zonder React? Gebruik dan de CSS en HTML die is beschreven in
```
